File: finance-manager-ui/src/app/dto/transaction-dto.ts
import { BaseDTO } from './base-dto';
import { TransactionCategoryDTO } from './transaction-category-dto';
import { BillDTO } from './bill-dto';

export class TransactionDTO extends BaseDTO {

    public type: boolean;
    public transactionCategory: TransactionCategoryDTO;
    public date: string;
    public description: string;
    public notes: string;
    public amount: number;
    public bill: BillDTO;

    // To be used only in UI.
    public enableDelete: boolean;

    constructor(date: string) {
        super();
        // Default values.
        this.type = true;
        this.amount = 0;
        this.enableDelete = false;  
        this.bill = new BillDTO(date);    
    }
}

File: finance-manager-server/src/main/java/com/github/cmag/financemanager/dto/TransactionDTO.java
package com.github.cmag.financemanager.dto;

import com.github.cmag.financemanager.config.AppConstants;
import com.github.cmag.financemanager.dto.master.data.TransactionCategoryDTO;
import com.github.cmag.financemanager.util.validation.NotAlreadyLinked;
import java.util.Date;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import lombok.Data;

/**
 * DTO representation of a Transaction.
 */
@Data
@NotAlreadyLinked
public class TransactionDTO extends BaseDTO {

  @NotNull(message = AppConstants.NOT_NULL)
  private TransactionCategoryDTO transactionCategory;

  @NotNull(message = AppConstants.NOT_NULL)
  private boolean type;

  @Size(max = 500, message = AppConstants.MAX_SIZE_500)
  private String notes;

  @NotEmpty(message = AppConstants.NOT_NULL)
  @Size(max = 250, message = AppConstants.MAX_SIZE_250)
  private String description;

  @NotNull(message = AppConstants.NOT_NULL)
  private double amount;

  @NotNull(message = AppConstants.NOT_NULL)
  private Date date;
  private BillDTO bill;
}
